### PR TITLE
core: limit message queue size

### LINKED ISSUE #2096: core: limit message queue size
Currently there can be up to 32k items in a message queue on 16bit platforms, and up to 2 billion items on 32 bit platforms. I don't think this is reasonable.

I would limit the message queue to 128 items. This reduces the size of `cib_t` (which is used in `tcb_t`) from 12 bytes to 4 bytes (including padding) on 32bit platforms.

If you think that 128 message queue items is too restrictive (I doubt it), then we can restrict the message queue to 32k items, which would shrink `cib_t` to 8 bytes.

But I don't think that the current limit of `MAXINT / 2` is reasonable. At least it hurts portability.

### PR DESCRIPTION
**Pull request by Darredevil**
fixes #2096

### CODE REVIEW

**Comment by Kijewski** [comment]
Why are you creating so many pull requests. Please read about the basics of Git! It's not hard ...

You did not fix https://github.com/RIOT-OS/RIOT/pull/2547#discussion-diff-26463606 in this iteration of your PR.

**Comment by Darredevil** [comment]
I apologize , I had a few problems with git and made a mess of a few PRs.
That specific PR #2547 was already labeled for squashing and I assumed it
was better to just close it.

I will be more careful in the future !



2015-03-22 18:36 GMT+02:00 René Kijewski <notifications@github.com>:

**Comment by Darredevil** [comment]
I understand some tests were failing because Travis CI notices i wrote "fix" in some commit messages, so i tried to rebase and change the commit messages, unfortunately it seems i also added an older commit regarding the header file include guards. The code itself is tested and works, my request would be for someone to merge it anyway even if Travis CI is still complaining about the commit messages. Thank you.

**Comment by LudwigOrtmann** [comment]
Sorry, you will have to fix the commit mess before this gets considered for merging.

**Comment by Darredevil** [comment]
Managed to clean up all that mess, awaiting approval from Travis CI.
Thanks for your patience :)

File: core/include/cib.h [hunk 1]
```diff
@@ -18,8 +18,10 @@
  * @author      unknown, propably Kaspar Schleiser <kaspar@schleiser.de>
  */
 
-#ifndef __CIB_H
-#define __CIB_H
+#include <stdint.h>
+
+#ifndef CIB_H
+#define CIB_H
```
Comment (LudwigOrtmann) [inline core/include/cib.h]: unrelated

**Comment by LudwigOrtmann** [comment]
You still did not address the comment by @Kijewski .

File: core/include/cib.h (hunk 1 above)
Comment (Darredevil) [inline core/include/cib.h]: I took the liberty to correct that too even if this isn't the main concern of this PR.

**Comment by Darredevil** [comment]
@LudwigOrtmann Forgot about that, changed the value to 2^15 as suggested by @Kijewski

**Comment by Darredevil** [comment]
@LudwigOrtmann Could you please remove the ```Needs Squashing``` label as it appears to mess with Travis's first test. Otherwise all looks good. Ready for merge.

**Comment by gebart** [comment]
@Darredevil The reason for the NEEDS SQUASHING label is that this PR should be squashed before being merged, in order to get rid of extra commits such as "Adjusted buffer size" in this PR. When the reviewer has ACKed the PR, they usually tell you what to squash, and then remove the label when they are satisfied with your commit list.

**Comment by Darredevil** [comment]
@LudwigOrtmann @gebart Squashed and ready for merge.

File: core/include/cib.h (hunk 1 above)
Comment (LudwigOrtmann) [inline core/include/cib.h]: Put it in a separate commit then. Check the coding conventions.

Comment (Darredevil) [inline core/include/cib.h]: Done.

File: core/include/cib.h [hunk 2]
```diff
@@ -71,12 +73,12 @@ static inline unsigned int cib_avail(cib_t *__restrict cib)
...
  */
-static inline int cib_get(cib_t *__restrict cib)
+static inline int16_t cib_get(cib_t *__restrict cib)
 {
-    unsigned int avail = cib_avail(cib);
+    uint16_t avail = cib_avail(cib);
 
     if (avail > 0) {
-        return (int) (cib->read_count++ & cib->mask);
+        return (int16_t) (cib->read_count++ & cib->mask);
```
Comment (authmillenon) [inline core/include/cib.h]: I'm unsure, if `sizeof(int16_t) == sizeof(uint16_t)` leads to problems here (though the original implementation obviously worked too).

Comment (Darredevil) [inline core/include/cib.h]: Followed the original implementation, tested locally and it passes all tests.

**Comment by Darredevil** [comment]
Travis finally agrees.

**Comment by authmillenon** [comment]
Ah sorry… I should stop working… did not see that the two required ACKs were still missing. :(

**Comment by authmillenon** [comment]
Sorry @Darredevil I think you have to open a new Pull Request. :(

**Comment by Darredevil** [comment]
:( Wouldn't it have been easier just to just that the patch is ok after merging instead of reverting?

**Comment by LudwigOrtmann** [comment]
No

**Comment by LudwigOrtmann** [comment]
Clicking the revert button and clicking the create PR button are easier than reviewing a core change.
